### PR TITLE
[FEAT] Add impl. to handle chunked encoding

### LINKED ISSUE #46: [DISCUSSION] Do we need the encoder like the decoder
@leske42 I wonder maybe we also need a encoder to encode it in different way, could you help me to check this? (bc it a little bit related to the builder)

_Originally posted by @LeaYeh in https://github.com/LeaYeh/webserver/issues/43#issuecomment-2333559821_

### PR DESCRIPTION
**Pull request by LeaYeh**
- [x] merge #85 first

Resolved #46 

I suggest run it and test it with browser.

### CODE REVIEW

File: include/defines.hpp [hunk 1]
```diff
@@ -250,4 +227,15 @@ enum EncodingMode
     GZIP = 0b00000010,
 };
 
+enum EventProcessingState
+{
+    INITIAL                 = 0b00000000,
+    PROCESSING              = 0b00001000,
+    WRITE_CHUNKED           = 0b00000100,
```
Comment (yenthing) [inline include/defines.hpp:234]: the order here is intentional?

Comment (LeaYeh) [reply include/defines.hpp:234]: Since uesd the bit map for the bit operations the order doesn't has meaning anymore only matters for readability concern.

**Review by yenthing — APPROVED** [review]

**Comment by leske42** [comment]
@LeaYeh this closes #46 right ?

File: include/defines.hpp [hunk 2]
```diff
@@ -226,4 +227,15 @@ enum EncodingMode
...
 };
 
+enum EventProcessingState
+{
+    INITIAL                 = 0b00000000,
+    PROCESSING              = 0b00000001,
+    WRITE_OTHERS_CHUNKED    = 0b00000010,
+    WRITE_FIRST_CHUNKED     = 0b00000100,
+    WRITE_CHUNKED           = 0b00001000,
+    COMPELETED              = 0b01000000,
```
Comment (leske42) [inline include/defines.hpp:237]: This should be "completed" but i mean i like this version too

File: source/shell/RequestAnalyzer.cpp [hunk 3]
```diff
@@ -103,12 +101,16 @@ RequestAnalyzerState RequestAnalyzer::state(void) const
 
 Request RequestAnalyzer::request(void) const
 {
-    std::cerr << "Request Line parsed. Method: " << _method << " Target: " << _uri.raw << " Version: " << _version << std::endl;
-    Request req; 
+    std::cerr << "Request Line parsed. Method: " << _method << "Target: " << _uri.raw << "Version: " << _version << std::endl;
```
Comment (leske42) [inline source/shell/RequestAnalyzer.cpp:104]: I want my space back

**Review by leske42 — CHANGES_REQUESTED** [review]

**Review by leske42 — APPROVED** [review]

**Pull request merged**